EventHubs/tests/test_eventhub_metrics.py

```diff
@@ -1,48 +1,101 @@
+import sys
 import unittest
-import datetime
+import time
+from io import StringIO
+from datetime import timedelta, datetime
 from baseeventhubtest import BaseEventHubTest
-
+from azure.monitor.query import LogsQueryClient, LogsQueryStatus
+from azure.mgmt.loganalytics import LogAnalyticsManagementClient
 
 class TestEventHubMetrics(BaseEventHubTest):
 
     def setUp(self):
         super(TestEventHubMetrics, self).setUp()
-        self.RESOURCE_GROUP_NAME = "TestEventHubMetrics-%s" % (
-            datetime.datetime.now().strftime("%d-%m-%y-%H-%M-%S"))
-        self.function_name_prefix = "EventHubs_Metrics"
-        self.STORAGE_ACCOUNT_NAME = "sumometlogs"
-        self.template_name = 'azuredeploy_metrics.json'
-        self.log_table_name = "AzureWebJobsHostLogs%d%02d" % (
-            datetime.datetime.now().year, datetime.datetime.now().month)
-        self.event_hub_namespace_prefix = "SumoMetricsNamespace"
-        self.eventhub_name = 'insights-metrics-pt1m'
-
+        self.RESOURCE_GROUP_NAME = "EventHubMetrics-%s" % (datetime.now().strftime("%d-%m-%y-%H-%M-%S"))
+        self.template_name = "azuredeploy_metrics.json"
+        self.event_hub_namespace_prefix = "SMNamespace"
+        self.eventhub_name = "insights-metrics-pt1m"
+        
     def test_pipeline(self):
-
         self.create_resource_group()
         self.deploy_template()
-        print("Testing Stack Creation")
-        self.assertTrue(self.resource_group_exists(self.RESOURCE_GROUP_NAME))
-        self.table_service = self.get_table_service()
-        self.insert_mock_logs_in_EventHub('metrics_fixtures.json')
-        self.check_error_logs()
+        self.assertTrue(self.resource_group_exists(self.RESOURCE_GROUP_NAME)) 
+        self.insert_mock_metrics_in_EventHub('metrics_fixtures.json')
+        time.sleep(300)  # Due to latency, logs are available after few mins.
+        self.check_success_log()
```

Comment:
to avoid 3 calls to fetch logs function we can fetch all the logs in one query 
app('sumometricsappinsights6dbfe7sr3obyi').traces
| where operation_Name == 'EventHubs_Metrics' 
| project operation_Id, timestamp, message, severityLevel

and filter the data using filter function in python code itself.
<img width="863" alt="image" src="https://github.com/SumoLogic/sumologic-azure-function/assets/3620468/19d16f33-c53c-4cfc-b2f8-99d24923619a">

Reply:
updated